### PR TITLE
Sliders move and reset

### PR DESCRIPTION
**Pull request by tcstewar**
Sliders which have a callable Node now move automatically.  You can also reset them to get back to the original function.

### CODE REVIEW

File: nengo_viz/components/slider.py [hunk 1]
```diff
@@ -36,13 +40,21 @@ def javascript(self):
                     label=self.label,
                     start_value=[float(x) for x in self.start_value])
         json = self.javascript_config(info)
-        return 'new VIZ.Slider(main, %s);' % json
+        return 'new VIZ.Slider(main, sim, %s);' % json
+
+    def update_client(self, client):
+        while len(self.data) > 0:
+            data = self.data.pop(0)
```
Comment (tbekolay) [inline nengo_viz/components/slider.py]: As noted in #148, this is O(n^2). I dunno if it really matters here, but regardless, should we do the same optimization as in that PR?

File: nengo_viz/static/viz_slider.js [hunk 2]
```diff
@@ -152,8 +164,8 @@ VIZ.Slider.prototype.constructor = VIZ.Slider;
 
 VIZ.Slider.prototype.set_value = function(slider_index, value, immediate) {
     console.assert(typeof slider_index == 'number');
-    console.assert(typeof value == 'number');
-    console.assert(typeof immediate == 'boolean');
+    console.assert(typeof value == 'number', value);
+    //console.assert(typeof immediate == 'boolean');
```
Comment (tbekolay) [inline nengo_viz/static/viz_slider.js]: Should this still be commented out?

File: nengo_viz/static/viz.js [hunk 3]
```diff
@@ -378,6 +378,12 @@ VIZ.DataStore.prototype.get_shown_data = function() {
     return shown;
 }
 
+VIZ.DataStore.prototype.is_at_end = function() {
+    var ts = this.sim.time_slider;
+    return (ts.last_time < ts.first_shown_time + ts.shown_time + 0.01);
```
Comment (tbekolay) [inline nengo_viz/static/viz.js]: Hmm... I'm a bit confused by this logic. I'm guessing that 0.01 is a fudge factor for float comparison; in that case, shouldn't it be something like
```javascript
ts.last_time >= ts.first_shown_time + ts.shown_time - 0.01
```
?

**Comment by tbekolay** [comment]
Made some comments inline with questions etc, but can confirm that this does indeed cause sliders to move! Pretty sweet.

File: nengo_viz/static/viz_slider.js (hunk 2 above)
Comment (tcstewar) [inline nengo_viz/static/viz_slider.js]: Argh, nope, I need to clear that up -- I added a third option, so it's not a boolean any more....  I'll add something to make that clearer.

File: nengo_viz/components/slider.py (hunk 1 above)
Comment (tcstewar) [inline nengo_viz/components/slider.py]: I still think there's the thread issue.  But here's an alternative:
```python
        count = len(self.data)
        if count > 0:
            for d in self.data[:count]:
                client.write(d, binary=True)
            del self.data[:count]
```

File: nengo_viz/static/viz.js (hunk 3 above)
Comment (tcstewar) [inline nengo_viz/static/viz.js]: Yup, that's for float fudge factor, but I made it way too large...   It does have to be a ```+```, though (with a - sign, the function can never return true).  +0.0000001 seems to work, though.

File: nengo_viz/static/viz_slider.js (hunk 2 above)
Comment (tcstewar) [inline nengo_viz/static/viz_slider.js]: Actually, that whole argument is going to disappear and get reorganized when #160 gets brought in.  So I'm tempted to just put a TODO in here.

Comment (tbekolay) [inline nengo_viz/static/viz_slider.js]: Yeah, or just remove the assert entirely, that's fine too.

**Pull request merged**